subgraphs/uniswap-v3/src/common/price/price.ts

```diff
@@ -139,46 +144,108 @@ export function findUSDPricePerToken(
       const pool = getLiquidityPool(poolAddress)!;
 
       if (pool.totalValueLockedUSD.gt(BIGDECIMAL_ZERO)) {
+        let whitelistTokenIndex = -1;
+        let tokenIndex = -1;
         if (pool.inputTokens[0] == token.id) {
-          // whitelist token is token1
-          const token1 = getOrCreateToken(event, pool.inputTokens[1], false);
-          // get the derived whitelist token in pool
-          const whitelistTokenValueLocked =
-            poolAmounts.inputTokenBalances[1].times(token1.lastPriceUSD!);
-          if (
-            whitelistTokenValueLocked.gt(largestWhitelistTokenValue) &&
-            whitelistTokenValueLocked.gt(
-              NetworkConfigs.getMinimumLiquidityThreshold()
-            )
-          ) {
-            largestWhitelistTokenValue = whitelistTokenValueLocked;
-            // token1 per our token * whitelist token per token1
-            priceSoFar = poolAmounts.tokenPrices[1].times(
-              token1.lastPriceUSD as BigDecimal
-            );
-          }
+          whitelistTokenIndex = 1;
+          tokenIndex = 0;
+        } else if (pool.inputTokens[1] == token.id) {
+          whitelistTokenIndex = 0;
+          tokenIndex = 1;
+        }
+
+        // Return if token is not in pool
+        if (tokenIndex == -1 || whitelistTokenIndex == -1) {
+          continue;
         }
-        if (pool.inputTokens[1] == token.id) {
-          const token0 = getOrCreateToken(event, pool.inputTokens[0], false);
-          // get the derived whitelist in pool
-          const whitelistTokenValueLocked =
-            poolAmounts.inputTokenBalances[0].times(token0.lastPriceUSD!);
-          if (
-            whitelistTokenValueLocked.gt(largestWhitelistTokenValue) &&
-            whitelistTokenValueLocked.gt(
-              NetworkConfigs.getMinimumLiquidityThreshold()
-            )
-          ) {
-            largestWhitelistTokenValue = whitelistTokenValueLocked;
-            // token0 per our token * whitelist token per token0
-            priceSoFar = poolAmounts.tokenPrices[0].times(
-              token0.lastPriceUSD as BigDecimal
-            );
+
+        // log.warning("HELLO", [])
+
+        // whitelist token is token1
```

Comment:
This comment is not accurate anymore since the index can be either 1 or 0?

Reply:
Correct! Removed.